File: examples/graphql-example/src/resources.rs
use mcp_factory_core::ResourceSpec;

pub fn build_resources() -> Vec<ResourceSpec> {
    vec![
        ResourceSpec {
            uri: "schema://graphql".to_string(),
            name: "graphql".to_string(),
            description: "Embedded GraphQL schema".to_string(),
            mime_type: "text/plain".to_string(),
            content: "type Query {\n  user(id: ID!): User\n}\n\ntype Mutation {\n  createUser(name: String!): User\n}\n\ntype User {\n  id: ID!\n  name: String!\n}\n",
        },
        ResourceSpec {
            uri: "meta://tools".to_string(),
            name: "tools".to_string(),
            description: "Generated tool index".to_string(),
            mime_type: "application/json".to_string(),
            content: "[\n  {\n    \"name\": \"user\",\n    \"description\": \"GraphQL query user\"\n  },\n  {\n    \"name\": \"createUser\",\n    \"description\": \"GraphQL mutation createUser\"\n  }\n]",
        },
    ]
}

File: generator/mcp_gen/resources.py
from __future__ import annotations

import json

from mcp_gen.models import ResourceSpec, ToolSpec


def build_embedded_resources(
    tools: list[ToolSpec],
    *,
    schema_uri: str,
    schema_name: str,
    schema_description: str,
    schema_mime_type: str,
    schema_text: str,
    tool_index_uri: str = "meta://tools",
    tool_index_name: str = "tools",
) -> list[ResourceSpec]:
    """Build the schema and generated-tool-index resources for one input."""
    return [
        ResourceSpec(
            uri=schema_uri,
            name=schema_name,
            description=schema_description,
            mime_type=schema_mime_type,
            content=schema_text,
        ),
        ResourceSpec(
            uri=tool_index_uri,
            name=tool_index_name,
            description="Generated tool index",
            mime_type="application/json",
            content=json.dumps(
                [{"name": tool.name, "description": tool.description} for tool in tools],
                indent=2,
            ),
        ),
    ]
